### DISCUSSION TITLE
async Gossipsub example useing Channels

### DISCUSSION THREAD

**Discussion by AronRath** [libp2p/rust-libp2p#2239] [Q&A] [answered]
Hello,

I'm new to Libp2p and was wondering if there is an example for an async implementation useing Gossipsub with communication through Channels.

I found one for Floodsub (https://github.com/thomaseizinger/rust-libp2p/blob/async-example/examples/chat-async.rs) but not sure how to change this to make it work with Gossipsub.

**Comment by mxinden** [comment, marked as answer]
Take a look at the below for an example on how to use gossipsub:

https://github.com/libp2p/rust-libp2p/blob/master/examples/gossipsub-chat.rs

See the below to learn how to integrate it into a larger application:

https://github.com/libp2p/rust-libp2p/blob/master/examples/file-sharing.rs
